rest/handler.go

```diff
@@ -267,6 +288,105 @@ func create2transferTx(w http.ResponseWriter, r *http.Request) {
 	_, _ = w.Write(output)
 }
 
+func txStatusHandler(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	txHash := params[KeyHash]
+	if len(txHash) == 0 {
+		WriteErrorResponse(w, http.StatusBadRequest, "TxHash not present")
+		return
+	}
+
+	tx, err := dbI.GetTxByHash(txHash)
+	if err != nil {
+		WriteErrorResponse(w, http.StatusBadRequest, "Unable to find transaction by hash")
+		return
+	}
+
+	output, err := json.Marshal(tx)
+	if err != nil {
+		WriteErrorResponse(w, http.StatusBadRequest, "Unable to marshall account")
+		return
+	}
+
+	// write headers and data
+	w.Header().Set("Content-Type", "application/json")
+	_, _ = w.Write(output)
+}
+
+// UserDetails is the body
+type UserDetails struct {
+	States    []UserDetailsState `json:"states"`
+	AccountID uint64             `json:"account_id"`
+	Pubkey    string             `json:"pubkey"`
+}
+
+type UserDetailsState struct {
+	Balance uint64 `json:"balance"`
+	StateID uint64 `json:"state_id"`
+	Token   uint64 `json:"token_id"`
+	Nonce   uint64 `json:"nonce"`
+}
+
+func userStateHandler(w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	pubkeyStr := params[KeyPubkey]
+	pubkeybz, err := hex.DecodeString(pubkeyStr)
+	if err != nil {
+		WriteErrorResponse(w, http.StatusBadRequest, "Could not decode string")
+		return
+	}
+
+	var response UserDetails
+	acc, err := dbI.GetAccountByPubkey(pubkeybz)
+	if err != nil {
+		WriteErrorResponse(w, http.StatusBadRequest, "Could not get account by pubkey")
+		return
+	}
+
+	response.AccountID = acc.AccountID
+	response.Pubkey = hex.EncodeToString(acc.PublicKey)
+
+	states, err := dbI.GetStateByAccID(acc.AccountID)
+	if err != nil {
+		WriteErrorResponse(w, http.StatusBadRequest, "Could not get state by accID")
+		return
+	}
+
+	var userStates []UserDetailsState
+	for _, state := range states {
+		if state.Type != core.TYPE_TERMINAL {
+			continue
+		}
+		_, balance, nonce, token, err := bazookaI.DecodeState(state.Data)
+		if err != nil {
+			WriteErrorResponse(w, http.StatusBadRequest, "Unable to decode state")
+			return
+		}
+		var stateData UserDetailsState
+		stateData.Balance = balance.Uint64()
+		stateData.Nonce = nonce.Uint64()
+		stateID, err := core.StringToUint(state.Path)
+		if err != nil {
+			fmt.Println("error converting path to ID")
```

Comment:
should use log and log the path that failed. so that the debug is easier.

Reply:
yeah let me handle all errors in another PR